core/state_processor.go

```diff
@@ -122,7 +110,12 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 
 	// Create a new receipt for the transaction, storing the intermediate root and gas used by the tx
 	// based on the eip phase, we're passing whether the root touch-delete accounts.
-	receipt := types.NewReceipt(root, result.Failed(), *usedGas)
+	var receipt *types.Receipt
+	if config.IsYoloV3(header.Number) {
+		receipt = types.NewEIP2718Receipt(tx.Type(), root, result.Failed(), *usedGas)
+	} else {
+		receipt = types.NewReceipt(root, result.Failed(), *usedGas)
+	}
```

Comment:
@fjl with the changes in how tx validation works now, shouldn't we be able to replace this entire thing with 
```
receipt = types.NewEIP2718Receipt(tx.Type(), root, result.Failed(), *usedGas)
```
Since the tx type is already verified earlier, and if it's included, we'll trust the type to be correct.